### PR TITLE
OpenTelemetry access log support formatter

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/model/telemetry_logging.go [hunk 1]
```diff
@@ -506,9 +509,43 @@ func buildOpenTelemetryAccessLogConfig(logName, hostname, clusterName, format st
...
+	// Otherwise, the configuration will be rejected.
+	cfg.Formatters = accessLogFormatters(format, labels)
+
 	return cfg
 }
 
+func accessLogFormatters(text string, labels *structpb.Struct) []*core.TypedExtensionConfig {
+	formatters := make([]*core.TypedExtensionConfig, 0, maxFormatterLength)
+	defer func() {
+		sort.Slice(formatters, func(i, j int) bool {
```
Comment (howardjohn) [inline pilot/pkg/model/telemetry_logging.go:523]: nit: slices.SortBy

File: releasenotes/notes/otel-accesslog-formatter.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: release-notes/v2
+kind: feature
+area: telemetry
+releaseNotes:
+  - |
+    **Added** access log formatter support for OpenTelemetry.
+upgradeNotes:
```
Comment (howardjohn) [inline releasenotes/notes/otel-accesslog-formatter.yaml:7]: This is not an upgrade note. An upgrade note means "when you upgrade, something is going to break".

This is saying "when you upgrade, you can now use a feature" which should be a release note

File: pilot/pkg/model/telemetry_logging.go [hunk 3]
```diff
@@ -62,6 +64,7 @@ const (
 	reqWithoutQueryCommandOperator = "%REQ_WITHOUT_QUERY"
 	metadataCommandOperator        = "%METADATA"
 	celCommandOperator             = "%CEL"
+	maxFormatterLength             = 3
```
[the remark this answers is not in the document]
Comment (zirain) [reply pilot/pkg/model/telemetry_logging.go:67]: count of all supported fotmatter, right now is 3.

File: pilot/pkg/model/telemetry_logging.go [hunk 4]
```diff
@@ -506,9 +509,43 @@ func buildOpenTelemetryAccessLogConfig(logName, hostname, clusterName, format st
...
+	cfg.Formatters = accessLogFormatters(format, labels)
+
 	return cfg
 }
 
+func accessLogFormatters(text string, labels *structpb.Struct) []*core.TypedExtensionConfig {
+	formatters := make([]*core.TypedExtensionConfig, 0, maxFormatterLength)
+	defer func() {
+		sort.Slice(formatters, func(i, j int) bool {
+			return formatters[i].Name < formatters[j].Name
```
[the remark this answers is not in the document]
Comment (zirain) [reply pilot/pkg/model/telemetry_logging.go:524]: it's just for stable/static slice order.

**Review by keithmattix — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**